galasa-managers-parent/galasa-managers-core-parent/dev.galasa.core.manager/src/main/java/dev/galasa/core/manager/internal/CoreManager.java

```diff
@@ -60,9 +72,51 @@ public void initialise(@NotNull IFramework framework, @NotNull List<IManager> al
 		}
 		this.ctf = framework.getConfidentialTextService();
 
+		if(galasaTest.isGherkin()) {
+			for(IGherkinExecutable gherkinExecutable : galasaTest.getGherkinTest().getAllExecutables()) {
+				for(Pattern regexPattern : gherkinPatterns) {
+					Matcher gherkinMatcher = regexPattern.matcher(gherkinExecutable.getValue());
+					if(gherkinMatcher.matches()) {
+						try {
+							gherkinExecutable.registerManager((IGherkinManager) this);
+						} catch (TestRunException e) {
+							throw new ManagerException("Unable to register Manager for Gherkin Statement", e);
+						}
+					}
+				}
+			}
+		}
+
 		// *** We always want the Core Manager initialised and included in the Test Run
 		activeManagers.add(this);
 	}
+	
+	@Override
+	public void executeGherkin(@NotNull IGherkinExecutable executable, Map<String, Object> testVariables)
+			throws ManagerException {
+		Matcher matcherStoreVariable = patternStoreVariable.matcher(executable.getValue());
+		if(matcherStoreVariable.matches()) {
```

Comment:
Create a static class to contain the execute method.   Otherwise the manager will explode